### PR TITLE
Support for Await

### PR DESCRIPTION
**Pull request by comphead**
Support for Await

### CODE REVIEW

**Comment by rssh** [comment]
Thanks a lot !  //will look today

**Comment by rssh** [comment]
Круто - смотрю
  //  я там еще гиттер вчера сделал: https://gitter.im/rssh/trackedfuture

**Comment by rssh** [comment]
btw, can you merge current master in this branch, to make pull-request fast-forward ?

File: agent/src/main/scala/trackedfuture/agent/MethodAdapter.scala [hunk 1]
```diff
@@ -2,77 +2,85 @@ package trackedfuture.agent
 
 import org.objectweb.asm._
 
-class TFMethodAdapter(up: MethodVisitor) extends MethodVisitor(Opcodes.ASM5, up) {
```
Comment (rssh) [inline agent/src/main/scala/trackedfuture/agent/MethodAdapter.scala:5]: If we rename TFMethodAdapter => MethodAdapter, why not rename TFClassAdapter => ClassAdapter ?
//of course, i can do this, just comment

File: agent/src/main/scala/trackedfuture/runtime/TrackedAwait.scala [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package trackedfuture.runtime
+
+import java.util.concurrent.TimeoutException
+
+import trackedfuture.util.ThreadLocalIterator
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, Awaitable}
+
+/**
```
Comment (rssh) [inline agent/src/main/scala/trackedfuture/runtime/TrackedAwait.scala:10]: Default IDE comments do not contain useful information.  (better create AUTHORS section in README)

File: agent/src/main/scala/trackedfuture/runtime/TrackedFuture.scala [hunk 3]
```diff
@@ -59,19 +61,19 @@ object TrackedFuture {
 
 
   def rmap[A, B](future: Future[A], function: A => B, executor: ExecutionContext): Future[B] = {
-    val trace = Thread.currentThread.getStackTrace()
```
Comment (rssh) [inline agent/src/main/scala/trackedfuture/runtime/TrackedFuture.scala:62]: we also can inject setCurrentFuture here  // ohh, wait - it's nontrivial, so better schedule for the next step

File: agent/src/main/scala/trackedfuture/runtime/TrackedFuture.scala [hunk 4]
```diff
@@ -59,19 +61,19 @@ object TrackedFuture {
...
 
   def rFlatMap[A, B](future: Future[A], function: A => Future[B], executor: ExecutionContext): Future[B] = {
-    val trace = Thread.currentThread.getStackTrace()
+    val trace = Thread.currentThread.getStackTrace
     val prevTrace = new StackTraces(trace, ThreadTrace.prevTraces.value)
     future.flatMap { a => trackedCall(function(a), prevTrace) }(executor)
   }
 
   def rFilter[A](future: Future[A], function: A => Boolean, executor: ExecutionContext): Future[A] = {
-    val trace = Thread.currentThread.getStackTrace()
```
Comment (rssh) [inline agent/src/main/scala/trackedfuture/runtime/TrackedFuture.scala]: and here ...

**Comment by comphead** [comment]
Ruslan is main logic done as expected? I'll fix denoted minors if it is

**Comment by rssh** [comment]
Yes. I prefer to change target exception, showing a trace of target future with message 'Trace of future waiting is ... '  but better do that as separate step

**Comment by rssh** [comment]
@comphead  - can you submit you patch again when done ?

**Comment by comphead** [comment]
yep, sure. I'll take a look on it at present Wed or Thu. Thanks

**Pull request merged**